File: tests/omp/collapse3.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  !! Sequence_basic.intro 1 [cFor "k"];
  !! Omp.parallel ~clause:[Num_threads "2"] [cVarDef "a"];
  !! Omp.for_  ~clause:[Collapse 2; Ordered_c 0;Private ["j"; "k"]; Schedule (Static, "3")] [cFor "k"];
  !! Omp.ordered [cFun "printf"];

)

File: tests/omp/collapse3.cpp
#include <omp.h>
#include <stdio.h>
void work(int a, int j, int k);
void sub()
{
  int  a;
  for (int k=1; k<=3; k++)
     for (int j=1; j<=2; j++)
     {
        printf("%d %d %d\n", omp_get_thread_num(), k, j);
        /* end ordered */
        work(a,j,k);
     }
}
